docs/adr/0007-allow-implementing-bindings-seperately.md

```diff
@@ -0,0 +1,98 @@
+
+**NOTE**: Version compatibility is handled in a [separate ADR](https://github.com/mozilla/uniffi-rs/pull/1203)
+
+## Considered Options
+
+* **[Option 1] Do nothing**
+
+    This means keeping everything as-is, and deciding that all binding generators (at least for now) should live under the `uniffi_bindgen` crate.
+
+* **[Option 2] Create a public API for external crates to implement their own bindings.**
+
+    Developers would have traits exposed they can leverage to implement binding generators that do not live in `uniffi_bindgen`. `uniffi_bindgen` would still handle generic tasks related to binding generation. This includes:
+
+    1. Parsing the UDL, and passing the `ComponentInterface` to the specific binding generator.
+    1. Parsing the `uniffi.toml` and passing the language specific configuration down to the binding generator.
+    1. Calling a generic `BindingGenerator::write_bindings` function that is implemented by the specific binding generator.
+    1. `uniffi_bindgen` would also expose its `CodeOracle`, `CodeType` and `CodeDeclaration` types to help developers create a standard way to interact with code generation. It shouldn't however, restrict developers to use those concepts if they don't fit a specific binding.
+
+## Pros and Cons of the Options
+
+### **[Option 1] Do nothing**
+
+* Good, because it makes it harder for users to accidentally use different versions of `uniffi` for scaffolding and bindings since they are all implemented together in the same crate.
+* Good, it makes it easier to make changes to multiple bindings at a time (in the case of a breaking change in `uniffi`, etc).
+* Bad, because maintainability can grow to be difficult - especially if more bindings are added which the core `uniffi` team is not familiar with.
+* Bad, because testability can also grow to be difficult - as more bindings are added the requirement to test all the bindings together in one repository is difficult to maintain.
+* Bad, because releases of all the binding generators are tied to the release of `uniffi_bindgen`.
+
+### **[Option 2] Create a public API for external crates to implement their own bindings.**
+
+* Good, because ownership will be clear, and members of the community can opt to maintain their own binding generators.
+* Good, because our CI would only need to test the core bindings we maintain, and others can be tested by their own maintainers (for example, a `gecko-js` binding generator should be tested in `mozilla-central` and not here).
+* Good, because a release in external bindings wouldn't have an impact on any internal ones unless it changes internal `uniffi` behavior.
+* Bad, because it's easier to accidentally have a version mismatch. (see ADR https://github.com/mozilla/uniffi-rs/pull/1203).
+* Bad, because testability increases in complexity. We are required to publish fixtures and examples we have. (see https://github.com/mozilla/uniffi-rs/pull/1206)
```

Comment:
maybe linkify this as \[PR 1026\](...) like done above?  (maybe the same above for the ADR as \[this ADR\]?)